modules/nw-running-dpdk-rootless-tap.adoc

```diff
@@ -0,0 +1,274 @@
+spec:
+ config: '{
+   "cniVersion": "0.4.0",
+   "name": "tap",
+   "plugins": [
+     {
+        "type": "tap",
+        "multiQueue": true,
+        "selinuxcontext": "system_u:system_r:container_t:s0",
+        "owner": 1001,
+        "group": 1001
+     },
+     {
+       "type":"tuning",
+       "capabilities":{
+         "mac":true
+       }
+     }
+   ]
+ }'
+----
+<1> Specify the same `target_namespace` where the `SriovNetwork` object is created.
+
+. Apply the `yaml` by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f tap-example.yaml
+----
+
+. Create a performance profile by following the procedure in "Creating a performance profile" section.
+
+. Create the following `Pod` spec, and then save the YAML in the `dpdk-pod-rootless.yaml` file:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dpdk-app
+  namespace: test-namespace <1>
+  annotations:
+    k8s.v1.cni.cncf.io/networks: '[
+      {"name": "sriov-network", "namespace": "test-namespace"},
+      {"name": "tap-one", "interface": "ext0", "namespace": "test-namespace"}]'
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: "worker-0"
+  securityContext:
+      fsGroup: 1001 <2>
+      runAsGroup: 1001 <3>
+      seccompProfile:
+        type: RuntimeDefault
+  containers:
+  - name: testpmd
+    image: <DPDK_image> <4>
+    securityContext:
+      capabilities:
+        drop: ["ALL"] <5>
+        add: <6>
+          - IPC_LOCK
+          - NET_RAW #for mlx only <7>
+      runAsUser: 1001 <8>
+      privileged: false <9>
+      allowPrivilegeEscalation: true <10>
+      runAsNonRoot: true <11>
+    volumeMounts:
+    - mountPath: /mnt/huge <12>
+      name: hugepages
+    resources:
+      limits:
+        openshift.io/sriovnic: "1" <13>
+        memory: "1Gi"
+        cpu: "4" <14>
+        hugepages-1Gi: "4Gi" <15>
+      requests:
+        openshift.io/sriovnic: "1"
+        memory: "1Gi"
+        cpu: "4"
+        hugepages-1Gi: "4Gi"
+    command: ["sleep", "infinity"]
+  runtimeClassName: performance-cnf-performanceprofile <16>
+  volumes:
+  - name: hugepages
+    emptyDir:
+      medium: HugePages
+----
++
+--
+<1> Specify the same `target_namespace` in which the `SriovNetwork` object is created. If you want to create the pod in a different namespace, change `target_namespace` in both the `Pod` spec and the `SriovNetwork` object.
+<2> Sets the group ownership of volume-mounted directories and files created in those volumes.
+<3> Specify the primary group ID used for running the container.
+<4> Specify the DPDK image that contains your application and the DPDK library used by application.
+<5> Removing all capabilities (`ALL`) from the container's securityContext means that the container has no special privileges beyond what is necessary for normal operation.
+<6> Specify additional capabilities required by the application inside the container for hugepage allocation, system resource allocation, and network interface access.
```

Comment:
We need to add a note here saying something like "Note that these capabilities must also be set in the binary file"

For example, adding these capabilities to dpdk-testpmd

RUN setcap cap_ipc_lock,cap_net_raw+ep /usr/local/bin/dpdk-testpmd

This is important for users to know.